crates/swift-bridge-build/src/package.rs

```diff
@@ -0,0 +1,287 @@
+//! Generate a Swift Package from Rust code
+
+use std::collections::HashMap;
+use std::fs;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+/// Config for generating Swift packages
+pub struct GeneratePackageConfig<'a> {
+	/// The directory containing the generated bridges
+	pub bridge_dir: &'a dyn AsRef<Path>,
+    /// Path per platform. e.g. `(ApplePlatform::iOS, "target/aarch64-apple-ios/debug/libmy_rust_lib.a")`
+	pub paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>,
+	/// The directory where the package will be saved
+	pub out_dir: &'a dyn AsRef<Path>,
+	/// The name for the Swift package
+	pub package_name: &'a str
+}
+
+impl<'a> GeneratePackageConfig<'a> {
+	/// Creates a new `GeneratePackageConfig` for generating Swift Packages from Rust code.
+	pub fn new(bridge_dir: &'a dyn AsRef<Path>, paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>, out_dir: &'a dyn AsRef<Path>, package_name: &'a str) -> Self {
+		Self {
+			bridge_dir,
+			paths,
+			out_dir,
+			package_name
+		}
+	}
+}
+
+#[allow(non_camel_case_types)]
```

Comment:
More of a personal preference, but would prefer using Rust style here. If you strongly disagree that's fine too. No big deal.